Add vibrancy setting for themes

ZhuZhengyi opened the pull request:
| Q                | A
| ---------------- | ---
| Bug fix?         | no
| New feature?     | yes
| BC breaks?       | no
| Deprecations?    | no
| New tests added? | not needed
| Fixed tickets    | comma-separated list of tickets fixed by the PR, if any
| License          | MIT

### Description

add vibrancy-dark theme for macos

#776 


--

#### Please, don't submit `/dist` files with your PR!

fxha commented on src/main/windows/editor.js:
```diff
@@ -69,6 +69,10 @@ class EditorWindow extends BaseWindow {
+    if (isOsx) {
+      win.setBackgroundColor('#00000000')
+      win.setVibrancy('ultra-dark')
+    }
```
@Jocs @ZhuZhengyi  I don't know about this feature but these values should only be set when the theme is active, right?

ZhuZhengyi replied:
> I tested this PR, it works well under the macOS system, but as @fxha said, it has no effect on Windows and Linux systems, and is consistent with one-dark, so I suggest that vibrancy-dark theme As a custom theme, we will support the custom theme in the v0.17.0. We also want to recommend vibrancy-dark as the official custom theme on our official website, but we may not merge this PR now, @ZhuZhengyi what's your opinion?

@Jocs ok!

ZhuZhengyi replied:
@fxha yes, setVibrancy only macOS electron API.

fxha commented on src/main/menu/templates/theme.js:
```diff
@@ -37,6 +37,14 @@ export default function (userPreference) {
+      checked: theme === 'vibrancy-dark',
+      click (menuItem, browserWindow) {
+        actions.selectTheme('vibrancy-dark')
+      }
```
@ZhuZhengyi The theme should only be available on macOS because on Linux and Windows we cannot set vibrancy. Meaning it's just One-Dark.

ZhuZhengyi replied:
Yes, it's based on one-dark.

Jocs commented:
I tested this PR, it works well under the macOS system, but as @fxha  said, it has no effect on Windows and Linux systems, and is consistent with one-dark, so I suggest that vibrancy-dark theme As a custom theme, we will support the custom theme in the v0.17.0. We also want to recommend vibrancy-dark as the official custom theme on our official website, but we may not merge this PR now, @ZhuZhengyi  what's your opinion?

ZhuZhengyi commented:
> I tested this PR, it works well under the macOS system, but as @fxha said, it has no effect on Windows and Linux systems, and is consistent with one-dark, so I suggest that vibrancy-dark theme As a custom theme, we will support the custom theme in the v0.17.0. We also want to recommend vibrancy-dark as the official custom theme on our official website, but we may not merge this PR now, @ZhuZhengyi what's your opinion?

@Jocs ok!

ZhuZhengyi commented:
> > I tested this PR, it works well under the macOS system, but as @fxha said, it has no effect on Windows and Linux systems, and is consistent with one-dark, so I suggest that vibrancy-dark theme As a custom theme, we will support the custom theme in the v0.17.0. We also want to recommend vibrancy-dark as the official custom theme on our official website, but we may not merge this PR now, @ZhuZhengyi what's your opinion?
> 
> @Jocs ok!

@Jocs @fxha, I removed vibrancy-dark theme and added vibrancy-type for offical themes, this can open vibrancy style for offical themes, and can select different vibrancy type on macOS.

Jocs commented:
Closed due to it's better to be implemented by custom theme in the later version.
